tools/js-tools/git-hooks/pre-push-hook.js

```diff
@@ -40,8 +41,69 @@ function checkChangelogFiles() {
 			)
 		);
 	} else {
+		const response = await promptChangelog();
+		if ( response ) {
+			try {
+				// Run the changelogger.
+				const autoChangelog = spawnSync( 'jetpack', [ 'changelog', 'add' ], {
+					stdio: 'inherit',
+				} );
+
+				// If the autochangelogger worked, commit the changelog files.
+				if ( autoChangelog.status === 0 ) {
+					const filesToCommit = [];
+					const changelogFiles = await spawnSync( 'git', [
+						'diff',
+						'--name-only',
+						'--diff-filter=A',
+						'--cached',
+					] )
+						.stdout.toString()
+						.trim()
+						.split( '\n' );
+
+					for ( const file of changelogFiles ) {
+						const match = file.match( /^projects\/([^/]+\/[^/]+)\/changelog\// );
+						if ( match ) {
+							filesToCommit.push( file );
+						}
+					}
+
+					if ( filesToCommit.length > 0 ) {
+						const commitFiles = await spawnSync( 'git', [
+							'commit',
+							...filesToCommit,
+							'-m',
+							'changelog',
+						] );
+						if ( commitFiles.status === 0 ) {
+							console.log(
+								chalk.green( 'Changelog file(s) committed! Go ahead and `git push` again.' )
+							);
+						}
+					}
+				}
+			} catch ( e ) {
+				console.log( 'Something went wrong', e );
+			}
+		}
 		process.exitCode = 1;
 	}
 }
 
+/**
+ * Prompts for for if we want to run the changelog automatically.
```

Comment:
Do we need to ask? If they refuse, they will get a failing test on their PR, so maybe it's better not to give them any choice? Especially since we ask again right after?

<img width="850" alt="image" src="https://user-images.githubusercontent.com/426388/158548700-4a95afc5-6616-464c-a712-b792cb77e4ac.png">

Reply:
>  so maybe it's better not to give them any choice? Especially since we ask again right after?

Agreed! I'll yank the prompt. I suppose the other option would be to let them continue pushing with `--no-verify` but see the note below.